sdk/include/opentelemetry/sdk/trace/tracer_factory.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include <memory>
+#include <vector>
+
+#include "opentelemetry/trace/tracer_factory.h"
+
+using opentelemetry::nostd::string_view;
+using opentelemetry::trace::Tracer;
+
+namespace opentelemetry
+{
+namespace sdk
+{
+namespace trace
+{
+
+class TracerFactory : public opentelemetry::trace::TracerFactory
+{
+public:
+  Tracer *const getTracer(const string_view &);
```

Comment:
I think this has to be virtual.  Please add formal parameters like "library_name" and "library_version". Any reason we can't use:
```
Tracer *const getTracer(const string_view &library_name, const string_view &library_version = "");
```

Reply:
Shouldn't the `virtual` on the API side of things handle the polymorphism requirement? Or are we expecting folks to subclass off the SDK classes as well? That said, I can't see a reason not to let them.

I'll add the names to the parameters.  Are there any strong opinions about variable naming abound?

Reply:
Nit: should string_view be passed by value rather than constref?

Reply:
> naming

[contributing.md](https://github.com/open-telemetry/opentelemetry-cpp/blob/master/CONTRIBUTING.md#development) &rarr; https://google.github.io/styleguide/cppguide.html#Naming

So I think `getTracer` should get capitalized to `GetTracer` but `library_name` is fine.

Reply:
> Nit: should string_view be passed by value rather than constref?
Don't really feel strongly about value vs const ref. Updated to pass by value instead, since it is more permissive for someone consuming the API.

Also updated the naming of everything to adhere to the linked guide, thanks for pointing that out!